### PR TITLE
Replaced !r with repr()

### PR DESCRIPTION
**Pull request by kvn219**
- [x] reference #29886
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

Issue: https://github.com/pandas-dev/pandas/issues/29886

@WillAyd

### CODE REVIEW

**Comment by WillAyd** [comment]
Thanks for the PR. Looks good and I think CI failures are unrelated / have been fixed on master since this was created.

To be sure, can you locally on your `fstring-sorting` branch run:

```sh
git fetch upstream
git merge upstream/master
git push origin fstring-sorting
```

I think should get CI green for this. Flag me down at PyData if you have any questions

**Review by simonjayhawkins — APPROVED** [review]
@kvn219 lgtm pending green ci

File: pandas/core/sorting.py [hunk 1]
```diff
@@ -264,7 +264,7 @@ def nargsort(items, kind="quicksort", ascending: bool = True, na_position="last"
     elif na_position == "first":
         indexer = np.concatenate([nan_idx, indexer])
     else:
-        raise ValueError("invalid na_position: {!r}".format(na_position))
+        raise ValueError(f"invalid na_position: {repr(na_position)}")
```
Comment (jreback) [inline pandas/core/sorting.py]: you can remove the repr on both of these as its unecessary

**Comment by WillAyd** [comment]
Looks like a merge conflict @kvn219 can you fix that up and re-push?

File: pandas/core/sorting.py (hunk 1 above)
Comment (kvn219) [reply pandas/core/sorting.py]: Thanks for the feedback @jreback, I just pushed with `repr()` removed.

**Comment by kvn219** [comment]
> Looks like a merge conflict @kvn219 can you fix that up and re-push?

**Comment by kvn219** [comment]
Hey @WillAyd, I'm not sure what to do. I'll go to HackLA next week and ask for help.

**Comment by simonjayhawkins** [comment]
Thanks @kvn219

**Pull request merged**